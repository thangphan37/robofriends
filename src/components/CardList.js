import React from 'react';
import Card from './Card';

function CardList({robots}) {
  return(
    <div className="tc">
      {robots.map((robot, index) => <Card key={index} robot={robot}/>)}
    </div>
  )
}

export default CardList;