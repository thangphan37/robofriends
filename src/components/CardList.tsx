import React from 'react';
import Card from './Card';

export interface Robot {
  id: string;
  username: string;
  email: string
}

const CardList: React.FC<{robots: Robot[]}> = ({robots}) => {
  return(
    <div className="tc">
      {robots.map((robot, index) => <Card key={index} robot={robot}/>)}
    </div>
  )
}

export default CardList;