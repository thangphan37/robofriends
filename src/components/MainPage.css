h1 {
  font-weight: 200;
  font-family: 'SEGA LOGO FONT';
  color: #0ccac4;
}