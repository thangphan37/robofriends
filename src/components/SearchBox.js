import React from 'react';

function SearchBox({onSearchRobots}) {
  return(
    <div className="pa2">
      <label htmlFor="robo" style={{color: '#FFFFFF', padding: '2px'}}>
        Let's type:
      </label>
      <input id="robo" type="search" className="pa3 ba b--green bg-lightest-blue" onChange={onSearchRobots} placeholder="Search robot"/>
    </div>
  )
}

export default SearchBox;