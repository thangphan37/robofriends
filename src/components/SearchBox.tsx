import React from 'react';

type Props = {
    onSearchRobots(event: React.FormEvent<HTMLInputElement>): void;
};

const SearchBox: React.FC<Props> = ({onSearchRobots}) => {
  return(
    <div className="pa2">
      <label htmlFor="robo" style={{color: '#FFFFFF', padding: '2px'}}>
        Let's type:
      </label>
      <input id="robo" type="search" className="pa3 ba b--green bg-lightest-blue" onChange={onSearchRobots} placeholder="Search robot"/>
    </div>
  )
}

export default SearchBox;