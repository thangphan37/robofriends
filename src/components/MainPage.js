import React from 'react';
import SearchBox from '../components/SearchBox';
import CardList from '../components/CardList';
import ScrollRobot from '../components/ScrollRobot';
import './MainPage.css';
import CounterButton from '../components/CountButton';

function MainPage({isPending, searchRobots, robots, searchField, requestRobots}) {
  React.useEffect(() => {
    requestRobots();
  }, []);

  function getRobots() {
    const newRobots = robots.filter(robot => robot.username.toLowerCase().includes(searchField.toLowerCase()));

    return newRobots;
  }

  return (isPending ? <h1>Loading...</h1>: (
    <div className="tc pa3">
      <h1 id="foo">ROBOFRIENDS</h1>
      <CounterButton />
      <SearchBox searchField={searchField} onSearchRobots={searchRobots}/>
      <ScrollRobot>
        <CardList robots={getRobots()}/>
      </ScrollRobot>
    </div>
  ))
}

export default MainPage;
