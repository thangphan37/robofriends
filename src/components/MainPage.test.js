import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import MainPage from './MainPage';
import SearchBox from './SearchBox';
import CardList from './CardList';

describe('test MainPage component', () => {
  let props;
  let useEffect;

  props = {
    robots: [{
      id: 1,
      username: 'john',
      email: 'john@gmail.com'
    }],
    isPending: false,
    searchField: '',
    requestRobots: jest.fn()
  };

  const mockUseEffect = () => {
    useEffect.mockImplementation(f => f());
  };

  beforeEach(() => {
    useEffect = jest.spyOn(React, 'useEffect');


    mockUseEffect();
  });

  it('snapshot MainPage component', () => {
    const wrapper = renderer.create(<MainPage {...props}/>).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('request Robots', () => {
    expect(props.requestRobots).toHaveBeenCalled();
  });

  it('search Robots', () => {
    const wrapper = shallow(<MainPage {...props}/>);

    expect(wrapper.find(CardList).prop('robots')).toEqual(props.robots);
  })
})

