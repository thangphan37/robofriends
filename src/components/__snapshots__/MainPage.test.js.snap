// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MainPage component snapshot MainPage component 1`] = `
<div
  className="tc pa3"
>
  <h1
    id="foo"
  >
    ROBOFRIENDS
  </h1>
  <div
    className="counter-container"
  >
    <button
      onClick={[Function]}
    >
      0
    </button>
  </div>
  <div
    className="pa2"
  >
    <label
      htmlFor="robo"
      style={
        Object {
          "color": "#FFFFFF",
          "padding": "2px",
        }
      }
    >
      Let's type:
    </label>
    <input
      className="pa3 ba b--green bg-lightest-blue"
      id="robo"
      placeholder="Search robot"
      type="search"
    />
  </div>
  <div
    style={
      Object {
        "border": "solid 5px #000000",
        "height": "500px",
        "overflowY": "auto",
      }
    }
  >
    <div
      className="tc"
    >
      <div
        className="pa3 grow bg-light-green ma2 tc bw2 br3 shadow-5 dib"
      >
        <img
          alt="robot"
          src="https://robohash.org/1?size=200x200"
        />
        <h2>
          john
        </h2>
        <p>
          john@gmail.com
        </p>
      </div>
    </div>
  </div>
</div>
`;
